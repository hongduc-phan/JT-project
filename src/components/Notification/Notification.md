Normal

```js
import Notification from './';
import {Notification as NotificationIcon} from '../Icons';

<Notification>
  <NotificationIcon className="notification" />
</Notification>;
```

Has notifications

```js
import Notification from './';
import {Notification as NotificationIcon} from '../Icons';

<Notification count={2}>
  <NotificationIcon className="notification" />
</Notification>;
```
