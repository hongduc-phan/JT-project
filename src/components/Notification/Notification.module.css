.notification {
  display: inline-flex;
  position: relative;
}

.badge {
  position: absolute;
  top: 0;
  right: 0;
  width: 1.125rem;
  height: 1.125rem;
  align-items: center;
  justify-content: center;
  text-align: center;
  display: inline-flex;
  font-size: 0.75rem;
  font-weight: var(--font-weight-bold);
  color: var(--white);
  background-color: var(--blue-dark);
  border-radius: 50%;
}
