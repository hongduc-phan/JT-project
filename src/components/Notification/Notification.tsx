import React, {ReactNode} from 'react';

import styles from './Notification.module.css';

export interface INotificationProps {
  children: ReactNode;
  count?: number;
}

const Notification = ({children, count = 0}: INotificationProps) => {
  return (
    <span className={styles.notification}>
      {0 < count && <span className={styles.badge}>{count}</span>}
      {children}
    </span>
  );
};

export default Notification;
