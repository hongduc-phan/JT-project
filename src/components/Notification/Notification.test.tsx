import React from 'react';
import renderer from 'react-test-renderer';
import {mount, ReactWrapper} from 'enzyme';

import Notification from './Notification';

describe('<Notification />', () => {
  describe('Normal', () => {
    let normalNotification: any;
    let component: ReactWrapper;

    beforeEach(() => {
      normalNotification = <Notification>Notification</Notification>;

      component = mount(normalNotification);
    });

    it('Should display a <span> tag', () => {
      const spans = component.find('span');
      expect(spans.length).toEqual(1);
    });

    it('Should match notification snapshot', () => {
      const tree = renderer.create(normalNotification).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Has notifications', () => {
    let normalNotification: any;
    let component: ReactWrapper;

    beforeEach(() => {
      normalNotification = <Notification count={2}>Notification</Notification>;

      component = mount(normalNotification);
    });

    it('Should display two spans', () => {
      const spans = component.find('span');
      expect(spans.length).toEqual(2);
    });

    it('Should display badge', () => {
      const spans = component.find('span.badge');
      expect(spans.length).toEqual(1);
    });

    it('Should match notification with bage snapshot', () => {
      const tree = renderer.create(normalNotification).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
