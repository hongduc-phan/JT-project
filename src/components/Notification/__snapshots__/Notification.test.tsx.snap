// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> Has notifications Should match notification with bage snapshot 1`] = `
<span
  className="notification"
>
  <span
    className="badge"
  >
    2
  </span>
  Notification
</span>
`;

exports[`<Notification /> Normal Should match notification snapshot 1`] = `
<span
  className="notification"
>
  Notification
</span>
`;
